### ISSUE TITLE
Incoherent scala-style typeclasses?

### ISSUE THREAD

**Issue by pshirshov** [flix/flix#6320]
Could you give me a hint if the language supports incoherent typeclasses?

In Scala my imports affect implementations and I can swap typeclass instances during compile-time and even in run-time (by importing from a `val`). 

Is anything like that possible in Flix?

**Comment by magnus-madsen** [comment]
No-- by design we do not want incoherence. For example, you only get one `Eq` instance per type.

Moreover, we resolve type classes at compile-time which is an enabler for optimizations.
